src/datasets.cpp

```diff
@@ -806,9 +804,10 @@ int SeedGenerator::new_seed()
      * (returns -1 if SeedGenerator is in non-deterministic mode).
      */
     if (this->meta_seed_==-1) {
-        return -1;
+        std::random_device rd;
+        return int(rd());
```

Comment:
Isn't this also returning a random number, thus meaning there's no difference in functionality between meta_seed_==-1 and !=-1? Per [docs](http://www.cplusplus.com/reference/random/random_device/operator()/), random_device rd() "Returns a new random number" between 0 and 4,294,967,295. Then Line 810 return this->distr_(this->eng_) is returning an int from uniform_int_distribution between 1 and 10,000. This may be the cause of the irreproducibility reported in #84 .

Reply:
We had a design choice:
1. If provided `meta_seed == -1`, the `SeedGenerator` constructor sets its `meta_seed_` to a random number acquired from the hardware. Otherwise, `meta_seed_ = meta_seed`. This seeds the `SeedGenerator` random engine with a specific seed; either the provided seed, or a random number. `new_seed()` then returns a random number from a uniform distribution `[1, 10000]` using the meta-seeded random generator.
2. The `SeedGenerator` constructor sets its `meta_seed_` property to equal the input `meta_seed`, even if it equals `-1`. In any case, it seeds its random number generator with that seed. However, `new_seed()` checks whether the set `meta_seed_ == -1` and, if it is, pulls a truly random number from the hardware. Otherwise, it returns a random number from the `[1, 10000]` uniform distribution using the seeded random number generator.

Essentially, 1. Handles the randomness in the constructor, whereas 2. handles it in `new_seed()`. They both produce the same results and are equally reproducible if given a `meta_seed != -1`.

@gpestre and I discussed, and we settled on 2. as the most transparent option. This way, when a `DecisionTree` is created and seeded with a `meta_seed = -1`, the tree's `SeedGenerator` has the attribute `meta_seed_ = -1`. Thus, while `SeedGenerator.new_seed()` produces truly random numbers to seed `df.sample()`, `DecisionTree.findBestSplit()` and the likes, the parent tree keeps the original seed it was given and it is absolutely clear that it is not seeded, and therefore uses random sub-seeds.

Reply:
I also made some new commits that fixed the issue, per my comment above. Forgot to add a comment to the PR, but this merged PR resolved #84 so that seeded `RandomForest` now produce consistent results.